snap/squashfs/squashfs.go

```diff
@@ -96,7 +96,7 @@ var runCommand = func(args ...string) error {
 	return err
 }
 
-func (s *Snap) unpack(src, dstDir string) error {
+func (s *Snap) Unpack(src, dstDir string) error {
```

Comment:
I pushed https://github.com/snapcore/snapd/pull/1665 that will allow non-root and uses the unapck in extractKernelAssets